src/test/java/org/jabref/logic/citationkeypattern/BracketedPatternTest.java

```diff
@@ -380,4 +380,52 @@ void expandBracketsWithTestCasesFromRegExpBasedFileFinder() {
                 "Eric von Hippel and Georg von Krogh have published Open Source Software and the \"Private Collective\" Innovation Model: Issues for Organization Science in Organization Science.",
                 BracketedPattern.expandBrackets("[author] have published [title] in [journal].", ',', entry, database));
     }
+
+    @Test
+    void expandBracketsWithFallbackToString() {
+        BibEntry bibEntry = new BibEntry()
+                .withField(StandardField.TITLE, "Title");
+
+        assertEquals("Title_eprint", BracketedPattern.expandBrackets("[title]_[EPRINT:(eprint)]", null, bibEntry, null));
+    }
+
+    @Test
+    void expandBracketsWithTwoFallbacks() {
+        BibEntry bibEntry = new BibEntry()
+                .withField(StandardField.YEAR, "2021");
+
+        assertEquals("2021", BracketedPattern.expandBrackets("[title:([EPRINT:([YEAR])])]", null, bibEntry, null));
```

Comment:
Please also ad test cases where the second and the fourth paramter are not null.